pkg/storegateway/bucket.go

```diff
@@ -1845,6 +1859,9 @@ type bucketBlock struct {
 	blockLabels labels.Labels
 
 	expandedPostingsPromises sync.Map
+
+	// Indicates whether the block is non-compacted and was queried.
+	queriedNonCompactedBlock bool
```

Comment:
It doesn't seem right to me to have this state on the `bucketBlock` for one specific use case, the block being queried isn't something intrinsic to the block.

Reply:
Updated the PR to remove this.

Reply:
What if we change this to just track whether a block was queried at all, and leave the non-compacted meaning out?